cmd/main.go

```diff
@@ -22,22 +22,32 @@ import (
 	"sigs.k8s.io/kubebuilder/v3/pkg/cli"
 	cfgv2 "sigs.k8s.io/kubebuilder/v3/pkg/config/v2"
 	cfgv3 "sigs.k8s.io/kubebuilder/v3/pkg/config/v3"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	basev1 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/base/v1"
 	declarativev1 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/declarative/v1"
-	pluginv2 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v2"
-	pluginv3 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v3"
+	golangv2 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v2"
+	golangv3 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v3"
 )
 
 func main() {
+
+	// Bundle plugin which built the golang projects scaffold by Kubebuilder
+	gov3Bundle, _ := plugin.NewBundle(golangv3.Plugin{}.Name(), golangv3.Plugin{}.Version(),
```

Comment:
Plugin and bundle should have different names. I know we are doing this in SDK but we already made clear that is "hackish" and we don't want to do the same here. The bundle should probably use "go.kubebuilder.io" while the plugin can use something like "base.go.kubebuilder.io". Delarative should probably also be renamed to "declarative.go.kubebuilder.io" as is it Golang-specific.

Reply:
IMO we should leave the `plugins`directory like this:
```
plugins/
  base/           --> "base.kubebuilder.io"
    v2/
    v3/
  golang/
    base/         --> "base.go.kubebuilder.io"
      v2/
      v3/
    declarative/  --> "declarative.go.kubebuilder.io"
      v1/

Reply:
I do not think that we need to do that. for v2. We can just provide the plugin base for v3 and leave v2 as it is and do not increase the effort to keep that maintained.

Reply:
There are 2 bases, `"base.kubebuilder.io"` (language-agnostic) will be the one scaffolding the `config` directory and `"base.go.kubebuiolder.io"` (go-specific) would be the golang specific files (`*_types.go`, `*_controller.go`, ...) so that we free up the `"go.kubebuilder.io"` name for the bundle. Names can be changed. The point is that the bundle and the plugin should have different names.

Related to the base language-agnostic plugin, I think we should od it for both v2 and v3, I don't think it will be harder to mantain. I actually think that it will be harder to mantain if we have to solve a bug for v2 and v3 if we have to search comething in the "base.kubebuilder.io/v1" and in "base.go.kubebuilder.io/v2"  plugins than if we just have to check both "base.kubebuilder.io" plugins.

Reply:
>There are 2 bases, "base.kubebuilder.io" (language-agnostic) will be the one scaffolding the config directory and "base.go.kubebuiolder.io" (go-specific) would be the golang specific files (*_types.go, *_controller.go, ...) so that we free up the "go.kubebuilder.io" name for the bundle. Names can be changed. The point is that the bundle and the plugin should have different names.

I do not agree with that because what is scaffold by the plugin in the config is not a BASE. It is their specific files that are generated according to the language plugin actions. 

Base == what equals no matter the language. Then, it does not necessary means only the config/ directory. it can be any thing that would be valid as a BASE for any language plugin. 

> Related to the base language-agnostic plugin, I think we should od it for both v2 and v3, I don't think it will be harder to mantain. I actually think that it will be harder to mantain if we have to solve a bug for v2 and v3 if we have to search comething in the "base.kubebuilder.io/v1" and in "base.go.kubebuilder.io/v2" plugins than if we just have to check both "base.kubebuilder.io" plugins.

Do we want people creating common bases with v2 for them to create their new plugins? E.g
I want to use kb only to generate the common base and then do the rest of the scaffolds using my implementation in java/python and etc.

I do not think so. We should push people to use the V3 layout and not the V2 plugin which only exist to keep the backwards compatibility with 2.x. However, we can always add that in the future if we have a real case that requires that. SDK will only consume the same base used for v3 too.

Reply:
Discussed offline to better understand ourselves.

Reply:
Thank you for your help and inputs. I think it is fine close to this one.